Struggling to compile code on macOS Sonoma 14.8.2

Hi,
Thank you for putting this repository together. I've been having a bit of an issue compiling the code with "pip install -e ." I've pasted the error message below in case you may have any insights:
```
(riscv_ise) cartergoldman@Carters-MacBook-Air-8 riscv_ise % pip install -e .        
Obtaining file:///Users/cartergoldman/sussex/knight_lab/riscv_ise
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [157 lines of output]
      /Library/Developer/CommandLineTools/usr/bin/make -C src/assembler
      /Library/Developer/CommandLineTools/usr/bin/make -C src/common
      make[1]: Nothing to be done for `all'.
      /Library/Developer/CommandLineTools/usr/bin/make -C genn/src/genn/genn
      make[1]: Nothing to be done for `all'.
      /Library/Developer/CommandLineTools/usr/bin/make -C src/compiler
      mkdir -p /Users/cartergoldman/sussex/knight_lab/riscv_ise/pyfenn
      c++ -shared -std=c++17 -march=native -Wall -Wpedantic -Wextra -MMD -MP -I/Users/cartergoldman/sussex/knight_lab/riscv_ise/include -fPIC -O3 -I/Users/cartergoldman/sussex/knight_lab/riscv_ise/genn/include/genn/genn -I/Users/cartergoldman/sussex/knight_lab/riscv_ise/genn/include/genn/third_party -I/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk/usr/include/ffi -o /Users/cartergoldman/sussex/knight_lab/riscv_ise/pyfenn/libcompiler_dynamic.so /Users/cartergoldman/sussex/knight_lab/riscv_ise/obj_dynamic/compiler//compiler.o /Users/cartergoldman/sussex/knight_lab/riscv_ise/obj_dynamic/compiler//process.o /Users/cartergoldman/sussex/knight_lab/riscv_ise/obj_dynamic/compiler//shape.o
      make[1]: Nothing to be done for `all'.
      /Library/Developer/CommandLineTools/usr/bin/make -C src/disassembler
      make[1]: Nothing to be done for `all'.
      /Library/Developer/CommandLineTools/usr/bin/make -C src/ise
      mkdir -p /Users/cartergoldman/sussex/knight_lab/riscv_ise/pyfenn
      Undefined symbols for architecture x86_64:
        "CodeGenerator::li(Reg, int)", referenced from:
            (anonymous namespace)::Visitor::visit(GeNN::Transpiler::Statement::Do const&) in compiler.o
            (anonymous namespace)::Visitor::visit(GeNN::Transpiler::Statement::While const&) in compiler.o
        "GeNN::Utils::scanCode(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&)", referenced from:
            NeuronUpdateProcess::NeuronUpdateProcess(ModelComponent::Private, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&, std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::shared_ptr<Parameter const>, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const, std::__1::shared_ptr<Parameter const>>>> const&, std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::shared_ptr<Variable const>, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const, std::__1::shared_ptr<Variable const>>>> const&, std::__1::unordered_map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::shared_ptr<EventContainer>, std::__1::hash<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::equal_to<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const, std::__1::shared_ptr<EventContainer>>>> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>> const&) in process.o
        "Error::Error(int)", referenced from:
            CodeGenerator::vlui(VReg, unsigned int) in compiler.o
            CodeGenerator::define_inner(std::__1::unordered_map<int, CodeGenerator::ClabelVal, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, CodeGenerator::ClabelVal>>>&, std::__1::unordered_multimap<int, CodeGenerator::Jmp, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, CodeGenerator::Jmp>>>&, int, unsigned int) in compiler.o
        "Label::~Label()", referenced from:
            (anonymous namespace)::Visitor::visit(GeNN::Transpiler::Statement::Do const&) in compiler.o
            (anonymous namespace)::Visitor::visit(GeNN::Transpiler::Statement::Do const&) in compiler.o
            (anonymous namespace)::Visitor::visit(GeNN::Transpiler::Statement::While const&) in compiler.o
            (anonymous namespace)::Visitor::visit(GeNN::Transpiler::Statement::While const&) in compiler.o
            (anonymous namespace)::Visitor::visit(GeNN::Transpiler::Statement::While const&) in compiler.o
            (anonymous namespace)::Visitor::visit(GeNN::Transpiler::Statement::While const&) in compiler.o
        "CodeGenerator::Jmp::appendCode(CodeGenerator*, unsigned int) const", referenced from:
            CodeGenerator::opJmp(Label const&, CodeGenerator::Jmp const&) in compiler.o
            CodeGenerator::opJmp(Label const&, CodeGenerator::Jmp const&) in compiler.o
        "CodeGenerator::Jmp::update(CodeGenerator*) const", referenced from:
            CodeGenerator::define_inner(std::__1::unordered_map<int, CodeGenerator::ClabelVal, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, CodeGenerator::ClabelVal>>>&, std::__1::unordered_multimap<int, CodeGenerator::Jmp, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, CodeGenerator::Jmp>>>&, int, unsigned int) in compiler.o
        "GeNN::Type::NumericValue::operator==(GeNN::Type::NumericValue const&) const", referenced from:
            GeNN::Type::ResolvedType::Numeric::operator==(GeNN::Type::ResolvedType::Numeric const&) const in process.o
            GeNN::Type::ResolvedType::Numeric::operator==(GeNN::Type::ResolvedType::Numeric const&) const in process.o
            GeNN::Type::ResolvedType::Numeric::operator==(GeNN::Type::ResolvedType::Numeric const&) const in process.o
        "typeinfo for Error", referenced from:
            CodeGenerator::vlui(VReg, unsigned int) in compiler.o
            CodeGenerator::define_inner(std::__1::unordered_map<int, CodeGenerator::ClabelVal, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, CodeGenerator::ClabelVal>>>&, std::__1::unordered_multimap<int, CodeGenerator::Jmp, std::__1::hash<int>, std::__1::equal_to<int>, std::__1::allocator<std::__1::pair<int const, CodeGenerator::Jmp>>>&, int, unsigned int) in compiler.o
        "_ffi_type_double", referenced from:
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_process.cc in process.o
        "_ffi_type_float", referenced from:
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_process.cc in process.o
        "_ffi_type_sint16", referenced from:
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            ...
        "_ffi_type_sint32", referenced from:
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_process.cc in process.o
            __GLOBAL__sub_I_process.cc in process.o
        "_ffi_type_sint8", referenced from:
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_process.cc in process.o
        "_ffi_type_uint16", referenced from:
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_process.cc in process.o
        "_ffi_type_uint32", referenced from:
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_process.cc in process.o
        "_ffi_type_uint64", referenced from:
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_process.cc in process.o
        "_ffi_type_uint8", referenced from:
            __GLOBAL__sub_I_compiler.cc in compiler.o
            __GLOBAL__sub_I_process.cc in process.o
      ld: symbol(s) not found for architecture x86_64
      c++ -shared -std=c++17 -march=native -Wall -Wpedantic -Wextra -MMD -MP -I/Users/cartergoldman/sussex/knight_lab/riscv_ise/include -fPIC -O3 -o /Users/cartergoldman/sussex/knight_lab/riscv_ise/pyfenn/libise_dynamic.so /Users/cartergoldman/sussex/knight_lab/riscv_ise/obj_dynamic/ise//dma_controller_sim.o /Users/cartergoldman/sussex/knight_lab/riscv_ise/obj_dynamic/ise//riscv.o /Users/cartergoldman/sussex/knight_lab/riscv_ise/obj_dynamic/ise//vector_processor.o
      clang++: error: linker command failed with exit code 1 (use -v to see invocation)
      make[1]: *** [/Users/cartergoldman/sussex/knight_lab/riscv_ise/pyfenn/libcompiler_dynamic.so] Error 1
      make: *** [compiler] Error 2
      make: *** Waiting for unfinished jobs....
      Undefined symbols for architecture x86_64:
        "getVOpType(unsigned int, unsigned int)", referenced from:
            VectorProcessor::calcOpResult(unsigned int, unsigned int, unsigned int, unsigned int, unsigned int) in vector_processor.o
        "getLoadType(unsigned int)", referenced from:
            RISCV::loadValue(unsigned int, int, unsigned int, unsigned int) const in riscv.o
        "getVMovType(unsigned int)", referenced from:
            VectorProcessor::executeInstruction(unsigned int, unsigned int (&) [32], ScalarDataMemory&, unsigned int) in vector_processor.o
        "getVSpcType(unsigned int)", referenced from:
            VectorProcessor::executeInstruction(unsigned int, unsigned int (&) [32], ScalarDataMemory&, unsigned int) in vector_processor.o
        "getVTstType(unsigned int)", referenced from:
            VectorProcessor::calcTestResult(unsigned int, unsigned int, unsigned int, unsigned int) const in vector_processor.o
        "getOpImmType(int, unsigned int)", referenced from:
            RISCV::calcOpImmResult(unsigned int, int, unsigned int, unsigned int) const in riscv.o
        "getStoreType(unsigned int)", referenced from:
            RISCV::executeStandardInstruction(unsigned int) in riscv.o
        "getVLoadType(unsigned int)", referenced from:
            VectorProcessor::executeInstruction(unsigned int, unsigned int (&) [32], ScalarDataMemory&, unsigned int) in vector_processor.o
        "getBranchType(unsigned int)", referenced from:
            RISCV::calcBranchCondition(unsigned int, unsigned int, unsigned int, unsigned int) const in riscv.o
        "getSystemType(int, unsigned int)", referenced from:
            RISCV::executeStandardInstruction(unsigned int) in riscv.o
        "getVOpImmType(int, unsigned int)", referenced from:
            VectorProcessor::calcOpImmResult(unsigned int, int, unsigned int, unsigned int) in vector_processor.o
        "getVStoreType(unsigned int)", referenced from:
            VectorProcessor::executeInstruction(unsigned int, unsigned int (&) [32], ScalarDataMemory&, unsigned int) in vector_processor.o
        "getOpType(int, unsigned int)", referenced from:
            RISCV::calcOpResult(unsigned int, unsigned int, unsigned int, unsigned int, unsigned int) const in riscv.o
      ld: symbol(s) not found for architecture x86_64
      clang++: error: linker command failed with exit code 1 (use -v to see invocation)
      make[1]: *** [/Users/cartergoldman/sussex/knight_lab/riscv_ise/pyfenn/libise_dynamic.so] Error 1
      make: *** [ise] Error 2
      Traceback (most recent call last):
        File "/Users/cartergoldman/opt/anaconda3/envs/riscv_ise/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 389, in <module>
          main()
        File "/Users/cartergoldman/opt/anaconda3/envs/riscv_ise/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 373, in main
          json_out["return_val"] = hook(**hook_input["kwargs"])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/cartergoldman/opt/anaconda3/envs/riscv_ise/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 209, in prepare_metadata_for_build_editable
          return hook(metadata_directory, config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/p6/cqsyhhv53g5978rhwcmgnr9m0000gn/T/pip-build-env-hzp7awo6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 486, in prepare_metadata_for_build_editable
          return self.prepare_metadata_for_build_wheel(
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/private/var/folders/p6/cqsyhhv53g5978rhwcmgnr9m0000gn/T/pip-build-env-hzp7awo6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 380, in prepare_metadata_for_build_wheel
          self.run_setup()
        File "/private/var/folders/p6/cqsyhhv53g5978rhwcmgnr9m0000gn/T/pip-build-env-hzp7awo6/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 317, in run_setup
          exec(code, locals())
        File "<string>", line 149, in <module>
        File "/Users/cartergoldman/opt/anaconda3/envs/riscv_ise/lib/python3.11/subprocess.py", line 413, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['make', 'backend', 'DYNAMIC=1', 'LIBRARY_DIRECTORY=/Users/cartergoldman/sussex/knight_lab/riscv_ise/pyfenn', '--jobs=2']' returned non-zero exit status 2.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> from file:///Users/cartergoldman/sussex/knight_lab/riscv_ise

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
(riscv_ise) cartergoldman@Carters-MacBook-Air-8 riscv_ise % 
```

--- a/src/MakefileCommon
+++ b/src/MakefileCommon
@@ -1,3 +1,6 @@
+# Detect OSX
+DARWIN				:=$(strip $(findstring DARWIN,$(OS_UPPER)))
+
 # Get directory of this makefile (means make can be invoked else where)
 RISCV_DIR:=$(abspath $(dir $(lastword $(MAKEFILE_LIST)))../)
 

--- a/src/MakefileCommon
+++ b/src/MakefileCommon
@@ -1,4 +1,5 @@
 # Detect OSX
+OS_UPPER			:=$(shell uname -s 2>/dev/null | tr [:lower:] [:upper:])
 DARWIN				:=$(strip $(findstring DARWIN,$(OS_UPPER)))
 
 # Get directory of this makefile (means make can be invoked else where)

--- a/src/assembler/Makefile
+++ b/src/assembler/Makefile
@@ -18,9 +18,16 @@ DEPS			:=$(OBJECTS:.o=.d)
 all: $(LIBASSEMBLER)
 
 ifdef DYNAMIC
+ifeq ($(DARWIN),DARWIN)
+$(LIBASSEMBLER): $(OBJECTS)
+	mkdir -p $(@D)
+	$(CXX) -dynamiclib -undefined dynamic_lookup $(CXXFLAGS) -o $@ $(OBJECTS)
+	install_name_tool -id "@loader_path/$(@F)" $@
+else
 $(LIBASSEMBLER): $(OBJECTS)
 	mkdir -p $(@D)
 	$(CXX) -shared $(CXXFLAGS) -o $@ $(OBJECTS)
+endif
 else
 $(LIBASSEMBLER): $(OBJECTS)
 	mkdir -p $(@D)

--- a/src/backend/Makefile
+++ b/src/backend/Makefile
@@ -21,9 +21,16 @@ CXXFLAGS +=-I$(RISCV_DIR)/genn/include/genn/genn -I$(RISCV_DIR)/genn/include/gen
 all: $(LIBBACKEND)
 
 ifdef DYNAMIC
+ifeq ($(DARWIN),DARWIN)
+$(LIBBACKEND): $(OBJECTS)
+	mkdir -p $(@D)
+	$(CXX) -dynamiclib -undefined dynamic_lookup $(CXXFLAGS) -o $@ $(OBJECTS)
+	install_name_tool -id "@loader_path/$(@F)" $@
+else
 $(LIBBACKEND): $(OBJECTS)
 	mkdir -p $(@D)
 	$(CXX) -shared $(CXXFLAGS) -o $@ $(OBJECTS)
+endif
 else
 $(LIBBACKEND): $(OBJECTS)
 	mkdir -p $(@D)

--- a/src/backend/backend_fenn.cc
+++ b/src/backend/backend_fenn.cc
@@ -626,6 +626,8 @@ private:
     class NeuronVarBase
     {
     public:
+        virtual ~NeuronVarBase() = default;
+
         //! Generate code to load vector register reg from memory before unrolled loop iteration r
         virtual void genLoad(::EnvironmentBase &env, VectorRegisterAllocator::RegisterPtr reg, uint32_t r) = 0;
         
@@ -888,6 +890,8 @@ private:
     class RowGeneratorBase
     {
     public:
+        virtual ~RowGeneratorBase() = default;
+
         RowGeneratorBase(CodeGenerator &c, std::shared_ptr<const EventPropagationProcess> process,
                          const Model::StateFields &stateFields,
                          ScalarRegisterAllocator &scalarRegisterAllocator, 

--- a/src/common/Makefile
+++ b/src/common/Makefile
@@ -18,9 +18,16 @@ DEPS			:=$(OBJECTS:.o=.d)
 all: $(LIBCOMMON)
 
 ifdef DYNAMIC
+ifeq ($(DARWIN),DARWIN)
+$(LIBCOMMON): $(OBJECTS)
+	mkdir -p $(@D)
+	$(CXX) -dynamiclib -undefined dynamic_lookup $(CXXFLAGS) -o $@ $(OBJECTS)
+	install_name_tool -id "@loader_path/$(@F)" $@
+else
 $(LIBCOMMON): $(OBJECTS)
 	mkdir -p $(@D)
 	$(CXX) -shared $(CXXFLAGS) -o $@ $(OBJECTS)
+endif
 else
 $(LIBCOMMON): $(OBJECTS)
 	mkdir -p $(@D)

--- a/src/compiler/Makefile
+++ b/src/compiler/Makefile
@@ -21,9 +21,16 @@ CXXFLAGS +=-I$(RISCV_DIR)/genn/include/genn/genn -I$(RISCV_DIR)/genn/include/gen
 all: $(LIBCOMPILER)
 
 ifdef DYNAMIC
+ifeq ($(DARWIN),DARWIN)
+$(LIBCOMPILER): $(OBJECTS)
+	mkdir -p $(@D)
+	$(CXX) -dynamiclib -undefined dynamic_lookup $(CXXFLAGS) -o $@ $(OBJECTS)
+	install_name_tool -id "@loader_path/$(@F)" $@
+else
 $(LIBCOMPILER): $(OBJECTS)
 	mkdir -p $(@D)
 	$(CXX) -shared $(CXXFLAGS) -o $@ $(OBJECTS)
+endif
 else
 $(LIBCOMPILER): $(OBJECTS)
 	mkdir -p $(@D)

--- a/src/disassembler/Makefile
+++ b/src/disassembler/Makefile
@@ -18,9 +18,16 @@ DEPS			:=$(OBJECTS:.o=.d)
 all: $(LIBDISASSEMBLER)
 
 ifdef DYNAMIC
+ifeq ($(DARWIN),DARWIN)
+$(LIBDISASSEMBLER): $(OBJECTS)
+	mkdir -p $(@D)
+	$(CXX) -dynamiclib -undefined dynamic_lookup $(CXXFLAGS) -o $@ $(OBJECTS)
+	install_name_tool -id "@loader_path/$(@F)" $@
+else
 $(LIBDISASSEMBLER): $(OBJECTS)
 	mkdir -p $(@D)
 	$(CXX) -shared $(CXXFLAGS) -o $@ $(OBJECTS)
+endif
 else
 $(LIBDISASSEMBLER): $(OBJECTS)
 	mkdir -p $(@D)

--- a/src/ise/Makefile
+++ b/src/ise/Makefile
@@ -18,9 +18,16 @@ DEPS			:=$(OBJECTS:.o=.d)
 all: $(LIBISE)
 
 ifdef DYNAMIC
+ifeq ($(DARWIN),DARWIN)
+$(LIBISE): $(OBJECTS)
+	mkdir -p $(@D)
+	$(CXX) -dynamiclib -undefined dynamic_lookup $(CXXFLAGS) -o $@ $(OBJECTS)
+	install_name_tool -id "@loader_path/$(@F)" $@
+else
 $(LIBISE): $(OBJECTS)
 	mkdir -p $(@D)
 	$(CXX) -shared $(CXXFLAGS) -o $@ $(OBJECTS)
+endif
 else
 $(LIBISE): $(OBJECTS)
 	mkdir -p $(@D)

